external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -636,4 +675,45 @@ public String toString() {
     private String getTopicsString() {
         return kafkaSpoutConfig.getSubscription().getTopicsString();
     }
+
+    public static class Info {
+        private final String topologyId;
+        private final int taskId;
+        private final String thread;
+
+        @JsonCreator
+        public Info(@JsonProperty("topologyId") String topologyId,
+                    @JsonProperty("taskId") int taskId,
+                    @JsonProperty("thread") String thread) {
+
+            this.topologyId = topologyId;
+            this.taskId = taskId;
+            this.thread = thread;
+        }
+
+        public Info(Thread thread, TopologyContext context) {
```

Comment:
Unless we're creating Infos in a lot of places, it might be better to remove this and use the other constructor instead. It makes it more obvious what's happening at the call site.

Reply:
Done